Index: Pdu.cpp
===================================================================
--- Pdu.cpp	(revision 737)
+++ Pdu.cpp	(working copy)
@@ -70,7 +70,7 @@
 
 unsigned short Pdu::getLength() const
 {
-    return _length;
+    return this->getMarshalledSize();
 }
 
 void Pdu::setLength(unsigned short pX)
@@ -95,7 +95,7 @@
     dataStream << _pduType;
     dataStream << _protocolFamily;
     dataStream << _timestamp;
-    dataStream << _length;
+    dataStream << this->getLength();
     dataStream << _padding;
 }
 
